packages/MainUI/components/Toolbar.tsx

```diff
@@ -0,0 +1,154 @@
+};
+
+export const Toolbar: React.FC<ToolbarProps> = ({ windowId, tabId }) => {
+  const { toolbar, loading } = useToolbar(windowId, tabId);
+
+  if (loading) {
+    return (
+      <Box display="flex" justifyContent="center" alignItems="center" height={64}>
+        Loading...
+      </Box>
+    );
+  }
+
+  const createToolbarConfig = () => {
+    const buttons = (toolbar as ToolbarResponse)?.response?.buttons || [];
+
+    return {
+      leftSection: {
+        buttons: buttons
+          .filter(btn => ['NEW', 'REFRESH', 'SAVE'].includes(btn.id))
+          .map(btn => {
+            const buttonConfig: ButtonConfig = {
+              key: btn.id,
+              icon: React.createElement(iconMap[btn.icon]),
+              tooltip: btn.name,
+              onClick: () => {},
+              disabled: !btn.enabled,
+              height: IconSize,
+              width: IconSize,
+              fill: theme.palette.baselineColor.neutral[0],
+            };
+
+            if (btn.id === 'NEW') {
+              buttonConfig.iconText = btn.name;
+            }
+
+            buttonConfig['sx'] = (() => {
+              if (btn.id === 'NEW') {
+                return {
+                  padding: '0.75rem',
+                  maxHeight: '2rem',
+                  background: theme.palette.baselineColor.neutral[100],
+                  borderRadius: '6.25rem 0 0 6.25rem',
+                  color: theme.palette.baselineColor.neutral[0],
+                  '&:hover': {
+                    background: theme.palette.dynamicColor.main,
+                  },
+                };
+              }
+              if (btn.id === 'SAVE') {
+                return {
+                  background: theme.palette.baselineColor.neutral[100],
+                  marginLeft: '0.2rem',
+                  border: `1px solid ${theme.palette.baselineColor.transparentNeutral[30]}`,
+                };
+              }
+              if (btn.id === 'REFRESH') {
+                return {
+                  padding: '0.75rem',
+                  maxHeight: '2rem',
+                  background: theme.palette.baselineColor.neutral[100],
+                  borderRadius: '0 6.25rem 6.25rem 0',
+                  color: theme.palette.baselineColor.neutral[0],
+                  '&:hover': {
+                    background: theme.palette.dynamicColor.main,
+                  },
+                };
+              }
+              return undefined;
+            })();
+
+            return buttonConfig;
+          }),
+        style: {
+          display: 'flex',
+          width: 'auto',
+          alignItems: 'center',
+          background: `var(--Neutral-0, ${theme.palette.baselineColor.neutral[0]})`,
+          borderRadius: '10rem',
+          padding: '0.25rem',
+          maxHeight: '2.5rem',
+          gap: '0.05rem',
+        },
+      },
+      centerSection: {
+        buttons: buttons
+          .filter(btn => ['DELETE', 'EXPORT', 'ATTACHMENTS'].includes(btn.id))
```

Comment:
It's a better way to define this array as constant

Reply:
👍🏼